File: LeetCode/1-100/66. Plus One/Solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int i=0;
        int size=digits.size();
        for(int j=size-1; j>=0; j--){
            if(digits[j]<9){
                digits[j]++;return digits;
            }
            else
                digits[j]=0;
        }
        
        vector<int> arr(size+1,0);
        arr[0]=1;
        return arr;
    }
};

File: LeetCode/1-100/66. Plus One/Solution.java
class Solution {
    public int[] plusOne(int[] digits) {
        int i = digits.length - 1;
        while (i >= 0) {
            if (digits[i] != 9) {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
            i--;
        }
        int[] arr = new int[digits.length + 1];
        arr[0] = 1;
        System.arraycopy(digits, 0, arr, 1, digits.length);
        return arr;
    }
}
